Binaries/dafny

```diff
@@ -13,14 +11,20 @@ done
 DAFNY_ROOT="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
 DAFNY="$DAFNY_ROOT/Dafny.exe"
 
-if [[ ! -x "$MONO" ]]; then
-    echo "Error: Dafny requires Mono to run on non-Windows systems."
-    exit 1
-fi
-
 if [[ ! -e "$DAFNY" ]]; then
     echo "Error: Dafny.exe not found at $DAFNY."
     exit 1
 fi
 
-"$MONO" "$DAFNY" "$@"
+MY_OS=$(uname -s)
+
+if [ "$(expr substr "$MY_OS" 1 5)" == "MINGW" ]; then
```

Comment:
Should we look for cygwin here too?  Or does no one use that anymore?

Reply:
If someone who uses Cygwin and can test this chimes in before this gets merged, great, else let's wait until someone actually wants the feature, I wouldn't want to put untested or unused stuff here.

Reply:
I use cygwin, well I'd prefer not to, but it's too late now.

Reply:
I use cygwin on my Windows machine at home.

Reply:
@RustanLeino are you using cygwin with mono-for-cygwin or are you using cygwin with direct invocation of Dafny.exe? Did the dafny script already work for you on cygwin without this change?